.github/workflows/scripts/auto-update-version.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash -e
+
+repo=$1
+variable_name=$2
+file_names=("${@:3}") # remaining args
+
+latest_version=$(gh api -q .tag_name "repos/open-telemetry/$repo/releases/latest" | sed 's/^v//')
+
+echo "Repo: $repo"
+echo "Latest version: $latest_version"
+
+for file_name in "${file_names[@]}"
+do
+  sed -i -e "s/$variable_name: .*/$variable_name: $latest_version/" "$file_name"
```

Comment:
It would be beneficial if this failed in the case that no match is found. It would help detect when a version field is (possibly) inadvertently rename or (re)moved from one of the target files. WDYT @trask?

Reply:
Is this a change you'll be adding in this or a followup PR @trask (or maybe have someone else do it)?

Reply:
ya, I'll add it in this PR

Reply:
done and tested 👍